src/main/java/data/TaskManager.java

```diff
@@ -0,0 +1,82 @@
+package data;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
+import static data.TaskManagerException.checkIfDateHasTasks;
+import static data.TaskManagerException.checkIfDateInCurrentWeek;
+
+
+public class TaskManager {
+    private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static final Map<LocalDate, List<String>> tasks = new HashMap<>();
+
+    public static void addTask(LocalDate date, String task) {
+        tasks.computeIfAbsent(date, k -> new ArrayList<>()).add(task);
+    }
+
+    public void deleteTask(LocalDate date, int taskIndex) {
+        List<String> dayTasks = tasks.get(date);
+        if (dayTasks != null && taskIndex >= 0 && taskIndex < dayTasks.size()) {
+            dayTasks.remove(taskIndex);
+            if (dayTasks.isEmpty()) {
+                tasks.remove(date);
+            }
+        }
+    }
+
+    public List<String> getTasksForDate(LocalDate date) {
+        return tasks.getOrDefault(date, new ArrayList<>());
+    }
+
+    public static void addManager(Scanner scanner, WeekView weekView, TaskManager taskManager)
+            throws TaskManagerException {
+        System.out.println("Enter the date for the task (dd/MM/yyyy):");
+        LocalDate date = getStringFromUser(scanner);
+
+        checkIfDateInCurrentWeek(date, weekView);
+
+        System.out.println("Enter the task description:");
+        String task = scanner.nextLine().trim();
+
+        addTask(date, task);
+        System.out.println("Task added.");
+    }
```

Comment:
After integration, we can invoke the UI class here instead of having the separate print statements

Reply:
We can shift all print statements and functions that deal with showing output and taking in input to the Ui Class!